ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,256 @@
+
+1. Parameters passed at the time of the trigger of a PipelineRun (that is user, git repository, etc)
+2. Output of upstream elements of the Pipeline which may be needed for performing the Task (that is build artifacts, container image, etc)
+3. Providing whether the Task was successful or not
+4. Output of the Task that may be needed for processing subsequent Tasks of the pipeline
+5. Output of the Task that need to be stored for auditing and troubleshooting (that is summarized results of a validation or a scan, etc)
+6. Communication with external systems that are needed for performing the Task (that is container image repository, image scanner, etc)
+7. Connection details and credentials that may be required for the above
+
+Related to [PLNSRVCE-41](https://issues.redhat.com/browse/PLNSRVCE-41) investigations in regards to [Tekton results](https://github.com/tektoncd/results). The investigations are documented [here](https://docs.google.com/document/d/1WGWx6MllVs-BwRLd0PP5QTQHgGp0fW2dOYRKRAKWiwo/edit#heading=h.bpyxfhyud4dv).
+
+## Decision
+
+### Results and Records
+
+The output of each Tekton task will be provided in two forms:
+- As a minimized [Tekton result](https://tekton.dev/docs/pipelines/tasks/#emitting-results) in JSON format listing all test failures
+  - The name of the result will be **HACBS_TEST_OUTPUT**
+- As a Test output JSON file with full test information that’s saved in the Tekton Pipeline Workspace
+  - The name of the file will be in snake case and will be in the form of **test_name_output.json**
+
+The maximum size of a [Task's Results](https://tekton.dev/vault/pipelines-v0.17.3/tasks/#emitting-results) is limited by the container [termination message](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/#customizing-the-termination-message) feature of Kubernetes.
+
+App Studio builds are structured as [a shared Persistent Volume per App Studio Workspace](https://docs.google.com/document/d/1IPlihVjkJ4Kb9tdhsk7iz3bn5rkT_SvJCNQhyXzK3aI/edit#bookmark=id.gefgys3vno2). This allows teams to share builds, implement caching and other shared volumes. A single persistent volume is mapped to each default build pipeline. Builds are passed a directory specific to their builds.
+
+#### Tekton Result Format
+
+The Test output of the Tekton result will be a JSON object that includes the context about the test along with the list of check names for all failed checks
+
+The output will provide the following information about the overall test result:
+- **result** - The outcome of the testing task, can be `SUCCESS`, `FAILURE`, `WARNING`, `SKIPPED` or `ERROR`
+- **namespace** - Optional, the rego namespace of the test policy
+- **timestamp** - An UNIX epoch timestamp of the test completion time
+- **successes** - The number of successful checks in the form of an integer
+- **note** - Optional, a short note providing additional information about the test
```

Comment:
Is the suggested use of this to describe the test itself or to pass information about the state of the test? The latter feels like it might have more utility but can also be used to fulfil the former (and stated) purpose.

All examples just have this as an empty string, but I would expect that to be omitted in those cases to save emitted result space.

Reply:
The `note` is never populated by the test itself, but always populated by the test stream, correct? If only used for `SKIPPED` results (and possibly for `ERROR` as discussed), maybe this would be more accurate so say that it is additional information about the test result?

If the field is indeed optional, I would expect it to be absent instead of having an empty string value. If some results have an empty string returned, I would expect all to have a minimum empty string. In this latter scenario, the key would not be optional, but additional context within the key is still optional.

Reply:
Yes, the content in `note` is provided by test stream. I remove the empty note from the examples
and change the description of `note` to `Optional, a short note provided by test workstream to provide additional information about the test`.

As mention above, [HACBS-1598](https://issues.redhat.com/browse/HACBS-1598) is filed to add a short message to [ERROR, SKIPPED, FAILURE] status result

Reply:
Thanks for filing that issue. I'll watch it!